lib/avoid_forbidden_namespaces.ex

```diff
@@ -0,0 +1,78 @@
+defmodule Nicene.AvoidForbiddenNamespaces do
+  @moduledoc """
+  Avoid calling forbidden namespaces. This is meant to be used for example
+  to avoid calling `AppWeb` from `App` in Phoenix applications.
+  It only works for top level namespaces.
+  """
+
+  @explanation [check: @moduledoc]
+
+  use Credo.Check, base_priority: :high, category: :refactoring
+
+  @doc false
+  def run(source_file, params \\ []) do
+    issue_meta = IssueMeta.for(source_file, params)
+
+    from_namespace = params |> Keyword.get(:from) |> get_namespace()
+
+    forbidden_namespaces =
+      params
+      |> Keyword.get(:forbid)
+      |> Enum.map(&get_namespace/1)
```

Comment:
That's a minor detail, but given both options are needed for check to work, making it crash on either missing seems better IMHO.

```suggestion
    from_namespace = params |> Keyword.fetch!(:from) |> get_namespace()

    forbidden_namespaces =
      params
      |> Keyword.fetch!(:forbid)
      |> Enum.map(&get_namespace/1)
```